clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/provider/agent/DockerRegistryImageCachingAgent.groovy

```diff
@@ -73,11 +78,13 @@ class DockerRegistryImageCachingAgent implements CachingAgent, AccountAware {
   }
 
   private Map<String, Set<String>> loadTags() {
-    credentials.repositories.collectEntries {
-      DockerRegistryTags tags = credentials.client.getTags(it)
-      tags ? [(tags.name): tags.tags ?: []] : [:]
+    credentials.repositories.findAll { it ->
+      threadCount == 1 || it.hashCode() % threadCount == index
```

Comment:
Technically `threadCount == 1` isn't needed, I just added it to avoid rehashing each repo when the comparison `it.hashCode() % threadCount == index` will always pass.